CRM/Event/Form/Participant.php

```diff
@@ -1340,20 +1342,22 @@ public function postProcess() {
         $contributionParams['currency'] = $config->defaultCurrency;
         $contributionParams['non_deductible_amount'] = 'null';
         $contributionParams['receipt_date'] = !empty($params['send_receipt']) ? CRM_Utils_Array::value('receive_date', $params) : 'null';
+        $contributionParams['contact_id'] = $this->_contactID;
+        // @todo change receive_date to datepicker field. Strip out all wrangling.
+        $contributionParams['receive_date'] = $params['receive_date'];
 
         $recordContribution = array(
-          'contact_id',
           'financial_type_id',
           'payment_instrument_id',
           'trxn_id',
           'contribution_status_id',
-          'receive_date',
           'check_number',
           'campaign_id',
+          'pan_truncation',
         );
 
         foreach ($recordContribution as $f) {
-          $contributionParams[$f] = CRM_Utils_Array::value($f, $params);
+          $contributionParams[$f] = CRM_Utils_Array::value($f, $this->_params);
```

Comment:
use of $this->_params includes the copied variables & reflects the fact the parent form & other classes use this. There is a tonne of confusion about the various arrays :-( I think we are moving back to $this->_params... at the moment